services/pull/check.go

```diff
@@ -118,13 +118,26 @@ func CheckPullMergeable(stdCtx context.Context, doer *user_model.User, perm *acc
 				err = nil
 			}
 
-			// * if the doer is admin, they could skip the branch protection check
+			// * if the doer is admin, they could sometimes skip the branch protection check
 			if adminSkipProtectionCheck {
-				if isRepoAdmin, errCheckAdmin := access_model.IsUserRepoAdmin(ctx, pr.BaseRepo, doer); errCheckAdmin != nil {
+				pb, pbErr := git_model.GetFirstMatchProtectedBranchRule(ctx, pr.BaseRepoID, pr.BaseBranch)
+				if pbErr != nil {
+					return fmt.Errorf("LoadProtectedBranch: %v", err)
```

Comment:
It seems not right. It should use `pbErr` but not `err`, and `Errorf` should use `%w` to wrap the error.

I will make some changes.